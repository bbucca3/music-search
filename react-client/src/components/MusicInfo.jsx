import React, { Component } from 'react';
import { Link } from 'react-router';

class MusicInfo extends Component {
  constructor(props) {
    super(props);
    // console.log('hello music props: ', props)
  }

  render() {
    return (
      <tr>
        <td><img src={this.props.albumArt} alt={this.props.trackName}/></td>
        <td>{this.props.artistName}</td>
        <td>{this.props.trackName}</td>
        <td>{this.props.albumName}</td>
        <td><Link to={{ pathname: '/lyrics',
                        query: {
                          artistName: this.props.artistName,
                          trackName: this.props.trackName,
                          albumName: this.props.albumName,
                          albumArt: this.props.albumArt
                        },
                        hash: {
                          artistName: this.props.artistName,
                          trackName: this.props.trackName,
                          albumName: this.props.albumName,
                          albumArt: this.props.albumArt
                        },
                        search: {
                          artistName: this.props.artistName,
                          trackName: this.props.trackName,
                          albumName: this.props.albumName,
                          albumArt: this.props.albumArt
                        }
                      }}
            >
              View Lyrics
            </Link>
        </td>
      </tr>
    )
  }
}

export default MusicInfo;
