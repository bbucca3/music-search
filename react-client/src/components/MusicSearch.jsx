import React, { Component } from 'react'
import $ from 'jquery';

import MusicInfo from './MusicInfo.jsx';

class MusicSearch extends Component {
  constructor(props) {
    super(props);
    this.state = {
      musicInput: "",
      musicList: [],
      resultCount: 0,
      showResults: false
    };
  }

  handleMusicInput() {
    // clear input for new search
    this.setState({musicList: []});

    // if I had more time here I would sanitize the input string
    // also I would take precautions on the server side xss injection
    
    // search for artists and songs
    $.ajax({
      type:'POST',
      url:'/api/musicsearch',
      data:{musicInput: this.state.musicInput},
      success:(results) => {
        console.log('music results ', results)
        let musicList = results.results.map((item, key) => {
          return <MusicInfo
                  key={item.trackId}
                  trackName={item.trackName}
                  artistName={item.artistName}
                  albumName={item.collectionName}
                  albumArt={item.artworkUrl60} />
        })
        console.log('musicList: ', musicList)
        this.setState({musicList, resultCount: results.resultCount, showResults: true})
      },
      error:(err) => {
        console.log('handleMusicInput ajax error: ', err);
      }
    })

  }

  render() {
    return (
        <div style={container}>
          <h3>Search for Musical Artist or Song: {this.state.musicInput}</h3>
          <div className="form-group">
            <input
              className="form-control"
              placeholder="enter search term"
              onChange={event => this.setState({musicInput: event.target.value})}
            />
            <button
              className="form-control"
              style={buttonStyle}
              onClick={() => this.handleMusicInput()}
            >
              Submit
            </button>
          </div>
          {this.state.showResults && <div>Results: {this.state.resultCount}</div>}
          {this.state.showResults && <div>
              <table style={{width:'100%'}}>
                <tbody>
                  <tr>
                    <th>Album Image</th>
                    <th>Artist Name</th>
                    <th>Track Name</th>
                    <th>Album Name</th>
                    <th>View Lyrics</th>
                  </tr>
                  {this.state.musicList}
                </tbody>
              </table>
            </div>}
        </div>
    );
  }
}

const container = {
  width: '90%',
  marginLeft: 'auto',
  marginRight: 'auto'
};

const buttonStyle = {
  textAlign: 'center',
  fontSize: '1rem',
  fontWeight: '200',
  background: 'transparent',
  borderRadius: '4px',
  margin: '10px 0',
  cursor: 'pointer',
  transition: 'all 0.3s ease',
  outline: 'none',
  border: '1px solid darkGreen',
  color: 'darkGreen'
};

export default MusicSearch;
