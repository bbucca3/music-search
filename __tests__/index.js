import React from 'react';

// components for testing
import MusicSearch from '../react-client/src/components/MusicSearch.jsx';
import MusicInfo from '../react-client/src/components/MusicInfo.jsx';

// Note: test renderer must be required after react
import renderer from 'react-test-renderer';

it('renders react', () => {
  const div = renderer.create(React.createElement('h6', null, 'ok'));
  expect(div.toJSON()).toMatchSnapshot();
})

it('renders MusicSearch correctly', () => {
  const div = renderer.create(React.createElement('div', null, <MusicSearch />));
  expect(div.toJSON()).toMatchSnapshot();
})

it('renders MusicInfo correctly', () => {
  const div = renderer.create(React.createElement('div', null, <MusicInfo />));
  expect(div.toJSON()).toMatchSnapshot();
})
