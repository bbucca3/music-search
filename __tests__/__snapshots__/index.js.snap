// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MusicInfo correctly 1`] = `
<div>
  <tr>
    <td>
      <img
        alt={undefined}
        src={undefined}
      />
    </td>
    <td />
    <td />
    <td />
    <td>
      <a
        onClick={[Function]}
        style={Object {}}
      >
        Lyrics
      </a>
    </td>
  </tr>
</div>
`;

exports[`renders MusicSearch correctly 1`] = `
<div>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
        "width": "90%",
      }
    }
  >
    <h3>
      Search for Musical Artist or Song: 
      
    </h3>
    <div
      className="form-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="enter search term"
      />
      <button
        className="form-control"
        onClick={[Function]}
        style={
          Object {
            "background": "transparent",
            "border": "1px solid darkGreen",
            "borderRadius": "4px",
            "color": "darkGreen",
            "cursor": "pointer",
            "fontSize": "1rem",
            "fontWeight": "200",
            "margin": "10px 0",
            "outline": "none",
            "textAlign": "center",
            "transition": "all 0.3s ease",
          }
        }
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;

exports[`renders react 1`] = `
<h6>
  ok
</h6>
`;
